harness/determined/exec/prep_container.py

```diff
@@ -28,9 +28,8 @@ def is_trial(info: det.ClusterInfo) -> bool:
 
 def download_context_directory(sess: api.Session, info: det.ClusterInfo) -> None:
     b64_tgz = bindings.get_GetTaskContextDirectory(sess, taskId=info.task_id).b64Tgz
-    if not is_trial(info) and len(b64_tgz) == 0:
+    if len(b64_tgz) == 0:
```

Comment:
I believe the code today ends up creating a gzip archive for the model definition in the table with no contents. it surfaced in the detached mode because detached mode didn't have the model definition either. as a result, IIRC the "download code" feature or "view code" tab had some issues, and webui had to blanket-disable these tabs for unmanaged experiments.

so, I'd say you should test how these features work on the web for your experiments w/o the model definition, cause I expect a bug there. 

perhaps it's time we did the right thing and made it nullable or even an FK'ed field: https://hpe-aiatscale.atlassian.net/browse/MLG-880

Reply:
I check the behavior on the CLI, webUI, and API:

CLI:
- returns an empty `.tgz` file

WebUI:
- Code tab in disabled (with hover-text: `Code file non-exist.`)
- No option to download the model_def

API 
- `experiments/<ID>/model_def`: redirects to overview page for experiment
- ``/experiments/<ID>/code``:  opens code page, but with an overlay that hangs on `Loading file tree...`; but can be clicked out of into different sections

Reply:
thank you for checking.